packages/core/src/internal/execution/types/execution-strategy.ts

```diff
@@ -216,7 +218,15 @@ export interface ExecutionStrategy {
    */
   name: string;
 
-  init: () => Promise<void>;
+  /**
+   * The configuration options for the strategy.
+   */
+  config: Record<PropertyKey, string | number> | Record<PropertyKey, never>;
+
+  init: (
+    deploymentLoader: DeploymentLoader,
+    jsonRpcClient: JsonRpcClient
+  ) => Promise<void>;
```

Comment:
This is internal and covers the current strategies. But basically the init should inject any services that a strategy might use.

Reply:
I'd say this is ok for now

Reply:
The only part that surprised me is that it now takes a wider type as argument when compared with the previous constructor. It used to use `LoadArtifactFunction` and now it's aware of the entire deployment.